Several BWC failures of MixedClusterClientYamlTestSuiteIT / search.highlight/20_fvh/Highlight query after release of 7.10.1

**Build scan**: https://gradle-enterprise.elastic.co/s/jracbd7txe7l2 / https://gradle-enterprise.elastic.co/s/eissbxg7vfko4 and many more

**Repro line**:

```
./gradlew ':qa:mixed-cluster:v6.8.14#mixedClusterTest' \
  -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" \
  -Dtests.method="test {p0=search.highlight/20_fvh/Highlight query}" \
  -Dtests.seed=3DBFFE73381C7536 \
  -Dtests.security.manager=true \
  -Dtests.bwc=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=en-AU \
  -Dtests.timezone=Mexico/BajaNorte \
  -Druntime.java=8
```

**Reproduces locally?**: yes

**Applicable branches**: 7.x / 7.10

**Failure history**:

https://gradle-enterprise.elastic.co/s/cfwfppp3jk5qs
https://gradle-enterprise.elastic.co/s/ze3k7c6oupzq6
https://gradle-enterprise.elastic.co/s/jracbd7txe7l2
https://gradle-enterprise.elastic.co/s/45bjefx3t6opu
https://gradle-enterprise.elastic.co/s/ukymgzhsuscuw
https://gradle-enterprise.elastic.co/s/ek4cgdwmj5p2k

**Failure excerpt**:

```
REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v6.8.14#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.highlight/20_fvh/Highlight query}" -Dtests.seed=3DBFFE73381C7536 -Dtests.security.manager=true -Dtests.bwc=true -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dtests.locale=en-AU -Dtests.timezone=Mexico/BajaNorte -Druntime.java=8
  2> java.lang.AssertionError: Failure at [search.highlight/20_fvh:60]: field [hits.hits.0.highlight.title.0] is null
        at __randomizedtesting.SeedInfo.seed([3DBFFE73381C7536:B5EBC1A996E018CE]:0)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:414)
        at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:391)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
        at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
        at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
        at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
        at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
        at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
        at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
        at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
        at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
        at java.lang.Thread.run(Thread.java:748)

        Caused by:
        java.lang.AssertionError: field [hits.hits.0.highlight.title.0] is null
            at org.junit.Assert.fail(Assert.java:88)
            at org.junit.Assert.assertTrue(Assert.java:41)
            at org.junit.Assert.assertNotNull(Assert.java:712)
            at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:79)
            at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:76)
            at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:407)
            ... 38 more
```

Relates 3cbe0eadce4a7d12247bbbc0899350dca7cea3b6 / 5e0b0c7414a6636a4ac82613564f3121250ccadd / d707f26eb9f7c1d649e04c84279e305038db133a

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/search.highlight/20_fvh.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/search.highlight/20_fvh.yml
@@ -1,7 +1,4 @@
 setup:
-  - skip:
-      version: "all"
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/66147"
   - do:
       indices.create:
         index: test
@@ -27,18 +24,20 @@ setup:
         index: test
         refresh: true
         body:
-          "title" : "The quick brown fox is brown"
-          "description" : "The quick pink panther is pink"
+          id : 1
+          title : "The quick brown fox is brown"
+          description : "The quick pink panther is pink"
 
   - do:
       index:
         index: test
         refresh: true
         body:
-          "title" : "The quick blue fox is blue"
-          "nested":
-            - "title": "purple octopus"
-            - "title": "purple fish"
+          id : 2
+          title : "The quick blue fox is blue"
+          nested:
+            - title: "purple octopus"
+            - title: "purple fish"
 
 ---
 "Highlight query":
@@ -46,6 +45,9 @@ setup:
       search:
         rest_total_hits_as_int: true
         body:
+          query:
+            term:
+              id: 1
           highlight:
             type: fvh
             fields:
@@ -80,6 +82,7 @@ setup:
             fields:
               title:
                 type: fvh
+          sort: [id]
 
   - match: {hits.hits.0.highlight.title.0: "The quick brown <em>fox</em> is brown"}
   - is_false: hits.hits.0.highlight.description

